### ISSUE TITLE
comparison of MFCC computation between librosa and essentia, for acoustic scene classification

### ISSUE THREAD

**Issue by edufonseca** [MTG/essentia#525]
A comparison was made of the MFCC computation between librosa and essentia, using data from [DCASE challenge 2016](http://www.cs.tut.fi/sgn/arg/dcase2016/task-acoustic-scene-classification), using their [baseline system](https://github.com/TUT-ARG/DCASE2016-baseline-system-python#dcase2016-baseline-system) (MFCC+GMM), for Task 1 - Acoustic scene classification.

**Procedure:**
1. Match common input parameters in both libraries and use same signal framing
2. Edited minor differences in librosa (20log10 and truncation of lowest amplitude values ) such that same amplitude treatment is used by both libraries 
3. Did not look into the filterbank (in theory, both based on Slaney’s)
4. Specific essentia params in [Windowing](http://essentia.upf.edu/documentation/reference/std_Windowing.html) algorithm: disable zero phase windowing, and leaving normalization as True (by default). Hence, the window normalization appears to be the only *major* difference between both computations, at least to the best of my knowledge.

**Run two simulations for Task 1 - Acoustic scene classification: with and without normalization.** 
Report the difference of **classification accuracy** found between librosa and essentia-based systems:
- Normalized = True -> accuracy difference ~ 6 % (librosa based system performs better)
- Normalized = False -> accuracy difference ~ +-0.3 %

Next plot shows the hamming window used in librosa and in essentia (Normalized = True). Note bottom of the plot.
![hamming_ess_zpwoff_normon](https://cloud.githubusercontent.com/assets/18170119/20844607/f2bcdc6c-b8c0-11e6-89a2-b63500b37bb0.png)

Next two plots show  mean and std of MFCCs computed over 1500 frames of the same audio file, for librosa and essentia. Up: with window normalization. Bottom: without window normalization
![mfcc_file7_small](https://cloud.githubusercontent.com/assets/18170119/20844752/a260cc1e-b8c1-11e6-85ee-72f957b97ea8.png)

![mfcc_file7_small_nfalse](https://cloud.githubusercontent.com/assets/18170119/20844768/babcf5e4-b8c1-11e6-9f0c-39dd6c444e78.png)

**Comment:** 
This occurs for this particular scenario, audio content (soundscapes) and classifier (GMM). Would something similar happen in a different scenario?

**Comment by dbogdanov** [comment]
I've created a separate issue concerning changes in MFCC values due to signal level #543. Normalized windowing will further contribute to this problem making mel energy values even smaller.

**Comment by dbogdanov** [comment]
We might want to change ```normalized``` to False by default.

**Comment by dbogdanov** [comment]
@edufonseca Do you still have your scripts to evaluate accuracy difference when using normalized windows again? (As we lowered the threshold for silence in #543, may be the normalization is not a problem any more).

**Comment by ChenJunHero** [comment]
I also  found that there were much differences of spectrum matrix between essentia and librosa.

**Comment by sildeag** [comment]
@edufonseca Did you compare with any other apps eg. OpenSmile, etc.?

**Comment by edufonseca** [comment]
No. Only with librosa. I think there is good chance that the differences
between librosa and essentia have been mitigated in later Essentia versions.
--
Eduardo Fonseca
Music Technology Group
Universitat Pompeu Fabra

--

**Comment by dbogdanov** [comment]
One of the main differences with librosa is in the silence threshold. We have done some updates related to that in the [mfcc_thresholding](https://github.com/MTG/essentia/tree/mfcc_thresholding) but it's not merged yet.   You can try to compare with MFCCs computed using that branch.
